### PR TITLE
test: remove side-effects from runtime builder

### PR DESCRIPTION
**Pull request by Stebalien**
Otherwise, reusing the builder can have strange side-effects when running multiple tests.

### CODE REVIEW

File: support/mock/builder.go [hunk 1]
```diff
@@ -13,16 +13,25 @@ import (
...
+		rt.receiver = receiver
+	})
+	return b
+}
+
+// Builds a new runtime object with the configured values.
+func (b RuntimeBuilder) Build(t testing.TB) *Runtime {
 	m := &Runtime{
-		ctx:               ctx,
+		ctx:               context.Background(),
```
Comment (acruikshank) [inline support/mock/builder.go:32]: This makes sense given the way the previous version behaved, but adding ctx as a parameter to `Build` could stave off some test problems in the future if this context is ever used.

**Review by acruikshank — APPROVED** [review]
Looks good. All the changes look seem to be big improvements.

File: support/mock/builder.go (hunk 1 above)
Comment (Stebalien) [reply support/mock/builder.go:32]: Hm. I agree. That's the right place to put the context.

Comment (Stebalien) [reply support/mock/builder.go:32]: Actually, I'm just going to make this construct the context internally and return a cancel function. That's the "least easy to mess up" solution.

Comment (Stebalien) [reply support/mock/builder.go:32]: That also avoids passing a confusing context. If I pass it as a parameter, it _looks_ like canceling the context should cancel the build operation.

Comment (Stebalien) [reply support/mock/builder.go]: Actually, I just dropped it all together. Instead, I cancel the context when the test ends.

**Pull request merged**